app/src/main/java/com/example/places/viewmodels/PlacesViewModel.kt

```diff
@@ -3,55 +3,72 @@ package com.example.places.viewmodels
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.example.places.PhotoResponse
-import com.example.places.Place
-import com.example.places.RetrofitInstance
+import com.example.places.*
 import kotlinx.coroutines.launch
+import retrofit2.Response
 
 class PlacesViewModel : ViewModel() {
     var placesLiveData = MutableLiveData<List<Place>>()
     var selectedPlace = MutableLiveData<Place>()
 
     fun getPlaces() {
-
         viewModelScope.launch {
-            val lisPhoto = mutableListOf<PhotoResponse?>()
-
+            val listPhoto = mutableListOf<PhotoResponse?>()
             val response = RetrofitInstance.api.getPlaces("41.8781,-87.6298", 9999)
+            val placesResponse = response.body()?.results
+            if (placesResponse != null) {
+                val listPlace = placesResponse.map {
+                    Place(
+                        it.fsq_id,
+                        it.name,
+                        it.location.address,
+                        it.email,
+                        it.description,
+                        it.tel,
+                        it.geocodes.main.latitude,
+                        it.geocodes.main.longitude
+                    )
+                }
+                placesLiveData.value = listPlace
+            }
+            getPhotoApi(listPhoto, response)
+        }
+    }
+
+    private fun getPhotoApi(
+        listPhoto: MutableList<PhotoResponse?>,
+        response: Response<PlaceResponse>
+    ) {
+        viewModelScope.launch {
             val responseBody = response.body()?.results
```

Comment:
This needs to be made more efficient. Add a comment here to remind us to work on this on Thursday.

`// TODO: <your comment here>` This will make a "to do" comment to keep us informed that something needs to be changed.

Reply:
// TODO : check this on Thursday